kartothek/io/dask/dataframe.py

```diff
@@ -301,3 +310,95 @@ def update_dataset_from_ddf(
+
+    .. warning::
+      If the size of the partitions is not evenly distributed, e.g. some partitions might be larger than others,
+      the metadata returned is not a good approximation for the whole dataset metadata.
+    .. warning::
+      Using the `frac` parameter is not encouraged for a small number of total partitions.
+
+
+    Parameters
+    ----------
+    store_factory
+      A factory function providing a KeyValueStore
+    dataset_uuid
+      The dataset's unique identifier
+    table_name
+      Name of the kartothek table for which to retrieve the statistics
+    predicates
+      Kartothek predicates to apply filters on the data for which to gather statistics
+
+      .. warning::
+          Filtering will only be applied for predicates on indices.
+          The evaluation of the predicates therefore will therefore only return an approximate result.
+
+    frac
+      Fraction of the total number of partitions to use for gathering statistics. `frac == 1.0` will use all partitions.
+    factory
+       A DatasetFactory holding the store and UUID to the source dataset.
+
+    Returns
+    -------
+    A pandas.DataFrame containing the following information about dataset statistics:
+       * `partition_label`: File name of the parquet file, unique to each physical partition.
+       * `row_group_id`: Index of the row groups within one parquet file.
+       * `row_group_byte_size`: Byte size of the data within one row group.
+       * `number_rows_total`: Total number of rows in one parquet file.
+       * `number_row_groups`: Number of row groups in one parquet file.
+       * `serialized_size`: Serialized size of the parquet file.
+       * `number_rows_per_row_group`: Number of rows per row group.
+
+    Raises
+    ------
+    ValueError
+      If no metadata could be retrieved, raise an error.
+
+    """
+    if not 0.0 < frac <= 1.0:
+        raise ValueError(
+            f"Invalid value for parameter `frac`: {frac}."
+            "Please make sure to provide a value larger than 0.0 and smaller than or equal to 1.0 ."
+        )
+    dataset_factory = _ensure_factory(
+        dataset_uuid=dataset_uuid,
+        store=store_factory,
+        factory=factory,
+        load_dataset_metadata=False,
+    )
+
+    mps = list(
+        dispatch_metapartitions_from_factory(dataset_factory, predicates=predicates)
+    )
+    if mps:
+        random.shuffle(mps)
+        # ensure that even with sampling at least one metapartition is returned
+        cutoff_index = max(1, int(len(mps) * frac))
+        mps = mps[:cutoff_index]
+        df = dd.from_delayed(
+            [
+                dask.delayed(MetaPartition.get_parquet_metadata)(
+                    mp, store=dataset_factory.store_factory, table_name=table_name,
+                )
+                for mp in mps
+            ]
+        )
+        df = df.compute()
```

Comment:
why compute? We usually only pass on the graphs. why do it differently herer?

Reply:
I usually try to avoid to use compute internally since compute also triggers optimization and behaves differently depending on whether or not there is a client present. Also, users may want to use `Client.compute(graph)` (lazy execution instead.